src/org/zaproxy/zap/extension/jsonView/HttpPanelJsonView.java

```diff
@@ -0,0 +1,174 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.jsonView;
+
+import java.awt.BorderLayout;
+
+import javax.swing.*;
+
+import net.sf.json.JSON;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+import org.apache.commons.configuration.FileConfiguration;
+import org.fife.ui.rtextarea.RTextScrollPane;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.httppanel.Message;
+import org.zaproxy.zap.extension.httppanel.view.*;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.request.RequestBodyStringHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.response.ResponseBodyStringHttpPanelViewModel;
+
+public class HttpPanelJsonView implements HttpPanelView, HttpPanelViewModelListener {
+
+	/**
+	 * Default name used for {@code MessageContainer}.
+	 *
+	 * @see org.zaproxy.zap.view.messagecontainer.MessageContainer
+	 */
+	public static final String DEFAULT_MESSAGE_CONTAINER_NAME = "HttpMessagePanel";
+	public static final String NAME = "HttpPanelJsonView";
+	private static final String CAPTION_NAME = "JSON";
+
+	private JTextArea httpPanelJsonArea;
+	private JPanel mainPanel;
+
+	private AbstractStringHttpPanelViewModel model;
+
+	public HttpPanelJsonView(AbstractStringHttpPanelViewModel model) {
+		//TODO disable line numbers, enable folding (requires rsyntaxtextview 2.6.1)
+		httpPanelJsonArea = new HttpPanelJsonArea();
+		RTextScrollPane scrollPane = new RTextScrollPane(httpPanelJsonArea);
+		scrollPane.setLineNumbersEnabled(false);
+		mainPanel = new JPanel(new BorderLayout());
+		mainPanel.add(scrollPane, BorderLayout.CENTER);
+		this.model = model;
+		model.addHttpPanelViewModelListener(this);
+	}
+
+	@Override
+	public void setSelected(boolean selected) {
+		if (selected) {
+			httpPanelJsonArea.requestFocusInWindow();
+		}
+	}
+
+	@Override
+	public String getName() {
+		return NAME;
+	}
+
+	@Override
+	public String getCaptionName() {
+		return CAPTION_NAME;
+	}
+
+	@Override
+	public String getTargetViewName() {
+		return "";
+	}
+
+	@Override
+	public int getPosition() {
+		return 0;
+	}
+
+	@Override
+	public boolean isEnabled(Message message) {
+		if(message == null){
+			return false;
+		}
+		if(message instanceof HttpMessage){
+			HttpHeader header;
+			if(this.model instanceof  RequestBodyStringHttpPanelViewModel){
+				header = ((HttpMessage) message).getRequestHeader();
+			} else if (this.model instanceof ResponseBodyStringHttpPanelViewModel){
+				header = ((HttpMessage) message).getResponseHeader();
+			} else {
+				return false;
+			}
+			if(header.getHeader("Content-Type") != null){
+				return header.getHeader("Content-Type").trim().contains("application/json");

```

Comment:
Could use `HttpHeader.CONTENT_TYPE` (in the getHeader call), you might want to force `toLower()` for the comparison as well (just incase).